pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -1091,13 +1085,44 @@ func BuildDefaultHTTPInboundRoute(clusterName string, operation string) *route.R
 	return val
 }
 
+func setTimeouts(action *route.RouteAction, defaultTimeout *duration.Duration,
+	vsTimeout *duration.Duration, node *model.Proxy,
+) {
+	// If default request timeout or virtual service timeout is not set, default to no timeout.
+	if defaultTimeout.AsDuration().Nanoseconds() == 0 && vsTimeout == nil {
```

Comment:
why need this separate check fo 0?

Reply:
cannot L1102-L1115 cover this case?

Reply:
Because we have to disable timeouts completely. and we can not set non zero maxstream in L1102 because of bug in Envoy. That is why we are using that deprecated field. This is much safer to do.

		action.MaxStreamDuration = &route.RouteAction_MaxStreamDuration{
			MaxStreamDuration: notimeout,
			// If not configured at all, the grpc-timeout header is not used and
			// gRPC requests time out like any other requests using timeout or its default.
			GrpcTimeoutHeaderMax: notimeout,
		}

Reply:
IC, could not treate`defaultTimeout` = `features.DefaultRequestTimeout` here.

Thet confuses me.

Why not separate inbound, which always has no timeout

Reply:
If it is confusing, I can separate inbound in to a separate function.